### ISSUE TITLE
can't group by a struct field, like ids->customerId

### ISSUE THREAD

**Issue by cameronbraid** [confluentinc/ksql#1849]
ksql 5.0

```
ksql> describe search_views;

Name                 : SEARCH_VIEWS
 Field            | Type                                                                                                                                                                                                                                                                                                                                                                                                                                             
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ROWTIME          | BIGINT           (system)                                                                                                                                                                                                                                                                                                                                                                                                                        
 ROWKEY           | VARCHAR(STRING)  (system)                                                                                                                                                                                                                                                                                                                                                                                                                        
 TIMESTAMP        | BIGINT                                                                                                                                                                                                                                                                                                                                                                                                                                           
 GUID             | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                                                                                                                                  
 ENVIRONMENT      | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                                                                                                                                  
 HOST             | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                                                                                                                                  
 RELEASE          | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                                                                                                                                  
 SOURCE           | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                                                                                                                                  
 TAG              | VARCHAR(STRING)                                                                                                                                                                                                                                                                                                                                                                                                                                  
 IDS              | STRUCT<CORRELATIONID VARCHAR(STRING), COUNTRYOFRESIDENCE VARCHAR(STRING), CUSTOMERID VARCHAR(STRING), CUSTOMERIPADDRESS VARCHAR(STRING), REQUESTID VARCHAR(STRING), SESSIONID VARCHAR(STRING), XSESSIONID VARCHAR(STRING)>                                                                                                                                                                                                                       
 SITECONTEXT      | STRUCT<COUNTRYCODE VARCHAR(STRING), DROPOFFDATE VARCHAR(STRING), DROPOFFLOCATION VARCHAR(STRING), LOCATIONCOUNTRYCODE VARCHAR(STRING), NITRO BOOLEAN, PAGE VARCHAR(STRING), PARAMS VARCHAR(STRING), PICKUPDATE VARCHAR(STRING), PICKUPLOCATION VARCHAR(STRING), PREFERREDVEHICLETYPE VARCHAR(STRING), SEARCHKEY VARCHAR(STRING), SEARCHTYPE VARCHAR(STRING), SITENAME VARCHAR(STRING), VEHICLE VARCHAR(STRING), VEHICLECATEGORY VARCHAR(STRING)> 
 SEARCHTOOLSMODEL | STRUCT<VIEW VARCHAR(STRING), SORTBY VARCHAR(STRING), SORTDIR VARCHAR(STRING), FILTER MAP<STRING, VARCHAR(STRING)>>                                                                                                                                                                                                                                                                                                                               
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
ksql> CREATE TABLE views_per_customer as select ids->customerId, count(*) from search_views WINDOW SESSION (60 seconds) group by ids->customerId;
name is null
ksql> 
```

**Comment by apurvam** [comment]
Hmm. interesting. Can you share the ksql server log for this run?

**Comment by miguno** [comment]
Any updates, @cameronbraid ?

**Comment by big-andy-coates** [comment]
KSQL doesn't currently support `GROUP BY` on a function call. Under the hood structs are handled by function calls... so are also not supported.  Working on a fix..

**Comment by rmoff** [comment]
@big-andy-coates I just hit this (group by struct doesn't work) - what release is the fix in? 5.0.1?

**Comment by apurvam** [comment]
The fix will be in 5.1. Only bug fixes will go into 5.0.x releases, and this is more like a feature addition.

**Comment by hjafarpour** [comment]
Fix moved to 5.2. It is available in master already.
